great_expectations/core/datasource_dict.py

```diff
@@ -108,7 +124,16 @@ def __getitem__(self, name: str) -> FluentDatasource | BaseDatasource:
 
         ds = self._datasource_store.retrieve_by_name(name)
         if isinstance(ds, FluentDatasource):
-            return self._init_fluent_datasource(ds)
+            hydrated_ds = self._init_fluent_datasource(ds)
+            if isinstance(hydrated_ds, SupportsInMemoryDataAssets):
+                for asset in hydrated_ds.assets:
+                    if asset.type == "dataframe":
+                        cached_data_asset = self._in_memory_data_assets.get(
+                            f"{name}-{asset.name}"
```

Comment:
Could we encapsulate this f-string name creation into a helper method? Same note about type comparison with enum

Reply:
I encapsulated the f-string into a method and created a new protocol for `InMemoryDataAsset`s.